state.ts

```diff
@@ -2,16 +2,32 @@ import AsyncStorage from "@react-native-async-storage/async-storage";
 import { create } from "zustand";
 import { createJSONStorage, persist } from "zustand/middleware";
 
+export type UserDataToSet = {
```

Comment:
Maybe we could use `Partial<UserData>` from typescript ([link](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype))